tests/abi/ABITests.cmake

```diff
@@ -0,0 +1,21 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+macro(register_abi_artifacts ABI ARCHITECTURE)
+
+register_artifact_category("reference_abi_binary_${ABI}" "ON")
+register_artifact("reference_abi_binary_${ABI}" "reference_binary" "${ARCHITECTURE}" "reference_binary.c")
+register_artifact_run("reference_abi_binary_${ABI}" "reference_binary" "default" "nope")
+
+register_artifact_category("runtime_abi_analysis_${ABI}" "ON")
+register_artifact("runtime_abi_analysis_${ABI}" "analyzed_binary" "${ARCHITECTURE}" "runtime_abi_analysis.c")
+register_artifact_run("runtime_abi_analysis_${ABI}" "analyzed_binary" "default" "nope")
```

Comment:
Not sure *analyzed* correctly captures what this binary does.
Maybe something more about "introspection"? But let's first decide whether it makes sense to merge `reference_binary.c` and `runtime_abi_analysis.c` or not.

Reply:
Totally agree - naming sucks but I never came op with anything better.

Reply:
Let's keep discussing in the other one.